### PR TITLE
Update 3.0.0 manifests to use JDK-23

### PR DESCRIPTION
**Pull request by reta**
### Description
Update 3.0.0 manifests to use JDK-23

### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/16255

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/3.0.0/opensearch-3.0.0-test.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ name: OpenSearch
 ci:
   image:
     name: opensearchstaging/ci-runner:ci-runner-al2-opensearch-build-v1
```
Comment (reta) [inline manifests/3.0.0/opensearch-3.0.0-test.yml:6]: @peterzhuamazon what would be the correct (latest?) CI runner image?

Comment (peterzhuamazon) [reply manifests/3.0.0/opensearch-3.0.0-test.yml:6]: You dont need to change this as we will always link the latest minor version to major version.

ex: ci-runner-al2-opensearch-build-v1 will always point to latest which is ci-runner-al2-opensearch-build-v1.4 that I just built yesterday.

**Comment by peterzhuamazon** [comment]
All images updated and sync now.

Thanks.

**Review by peterzhuamazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Trigger new 3.0.0 build run on OpenSearch with JDK23: https://build.ci.opensearch.org/job/distribution-build-opensearch/10388/console

**Comment by reta** [comment]
Awesome, thanks a lot @peterzhuamazon !

**Pull request merged**